Source/DataModels/Windows/EditorWindows/WindowInspector.cpp

```diff
@@ -149,6 +149,13 @@ void WindowInspector::InspectSelectedGameObject()
 				}
 			}
 
+			if (!lastSelectedGameObject->GetComponent(ComponentType::ANIMATION)) {
+				if (ImGui::MenuItem("Create Animation Component"))
+				{
+					AddComponentAnimation();
```

Comment:
```suggestion
					App->scene->GetSelectedGameObject()->CreateComponent(ComponentType::ANIMATION);
```
you can delete the function at 357

Reply:
we are encapsulating into methods all the logic related to components. There are other methods such as AddComponentPlayer, AddComponentRigidBody and AddComponentMockState that are defined in a similar way. I think that it's good to have it like this for reutilisation purposes. I would suggest to do a refactor in the future if we don't find this useful